### PR TITLE
Test Fixture ExplicitRoot + ModuleResolutionRelativePathOutsideRoot.

### PR DESCRIPTION
**Pull request by rickvanprim**
Updates `module_resolution_relative_path_outside_root` test to check valid paths outside of the root, by moving the root to a subpath so that paths outside of it are possible.  If this would be more appropriate as a new test, or if the original check for an invalid path should be left, I'm happy to update.

### CODE REVIEW

File: crates/hir_def/src/nameres/tests/mod_resolution.rs [hunk 1]
```diff
@@ -323,13 +323,26 @@ pub struct Baz;
 fn module_resolution_relative_path_outside_root() {
     check(
         r#"
-//- /main.rs
+//- /a/b/c/d/e/main.rs root:
```
Comment (matklad) [inline crates/hir_def/src/nameres/tests/mod_resolution.rs:326]:
```suggestion
//- /a/b/c/d/e/main.rs crate:main
```

I think this should do the same without introducnig code for `explicit_root` handling

File: crates/hir_def/src/nameres/tests/mod_resolution.rs [hunk 2]
```diff
@@ -323,13 +323,26 @@ pub struct Baz;
...
 #[path="../../../../../outside.rs"]
 mod foo;
+
+//- /outside.rs
+mod bar;
+
+//- /bar.rs
```
Comment (matklad) [inline crates/hir_def/src/nameres/tests/mod_resolution.rs:333]: Hm, shouldn't it be `/outside/bar.rs`? I see that the test passes, so there might be a bug in rust-analyzer?

Comment (rickvanprim) [reply crates/hir_def/src/nameres/tests/mod_resolution.rs:333]: Good catch, you're right that should be `/outside/bar.rs` or `/outside/bar/mod.rs`.  I'll see if I can figure out what's happening there.

File: crates/hir_def/src/nameres/tests/mod_resolution.rs (hunk 1 above)
Comment (rickvanprim) [reply crates/hir_def/src/nameres/tests/mod_resolution.rs:326]: That worked.  I'll revert the explicit root stuff.

File: crates/hir_def/src/nameres/tests/mod_resolution.rs (hunk 2 above)
Comment (rickvanprim) [reply crates/hir_def/src/nameres/tests/mod_resolution.rs:333]: Err actually I think this behavior is correct, my bad.  If you declare a non-inline mod, it's located in the same parent as where it's declared.  `/main.rs: mod foo;` -> `/foo.rs`

**Pull request merged**